l2geth/rollup/l2_gas_pricer.go

```diff
@@ -0,0 +1,43 @@
+package rollup
+
+import (
+	"math"
+)
+
+type GetTargetGasPerSecond func() float64
+
+type L2GasPricer struct {
+	curPrice                 float64
+	floorPrice               float64
+	getTargetGasPerSecond    GetTargetGasPerSecond
+	maxPercentChangePerEpoch float64
+}
+
+func NewGasPricer(curPrice float64, floorPrice float64, getTargetGasPerSecond GetTargetGasPerSecond, maxPercentChangePerEpoch float64) *L2GasPricer {
+	return &L2GasPricer{
+		curPrice:                 math.Max(curPrice, floorPrice),
+		floorPrice:               floorPrice,
+		getTargetGasPerSecond:    getTargetGasPerSecond,
+		maxPercentChangePerEpoch: maxPercentChangePerEpoch,
+	}
+}
+
+// Calculate the next gas price given some average gas per second over the last epoch
+func (p *L2GasPricer) CalcNextEpochGasPrice(avgGasPerSecondLastEpoch float64) float64 {
+	avgGasPerSecondLastEpoch = math.Max(0, avgGasPerSecondLastEpoch)
+	// The percent difference between our current average gas & our target gas
+	proportionOfTarget := avgGasPerSecondLastEpoch / p.getTargetGasPerSecond()
```

Comment:
Perhaps we should have defined behavior + a test case for when `getTargetGasPerSecond` returns `0`

```go
package main

import (
	"fmt"
)

func main() {
	a := float64(0)
	b := float64(100)
	
	c := b / a
	fmt.Println(c)
}
```

Running the above code prints `+Inf`